contracts/fee_splitter/src/testing.rs

```diff
@@ -1 +1,174 @@
+use bindings_test::{mock_coreum_deps, CoreumApp};
+use coreum_wasm_sdk::core::CoreumMsg;
+use cosmwasm_std::{
+    testing::{mock_env, mock_info, MOCK_CONTRACT_ADDR},
+    Addr, Attribute, BankMsg, Coin, CosmosMsg, Decimal, ReplyOn, Uint128,
+};
+use cw_multi_test::{ContractWrapper, Executor};
 
+use crate::{
+    contract::{execute, instantiate},
+    msg::{ExecuteMsg, InstantiateMsg, QueryMsg},
+    state::Config,
+};
+
+pub type SubMsg = cosmwasm_std::SubMsg<CoreumMsg>;
+
+#[test]
+fn init_works() {
+    let mut app = CoreumApp::default();
+
+    let code_id = store_fee_splitter_code(&mut app);
+    let sender = "addr0000";
+
+    let first_tupple = ("tokenA".to_string(), Decimal::from_ratio(1u128, 2u128));
+    let second_tuple = ("tokenB".to_string(), Decimal::from_ratio(1u128, 2u128));
+    let msg = InstantiateMsg {
+        addresses: vec![first_tupple.clone(), second_tuple.clone()],
+        cw20_contracts: vec!["cw20_contract_one".to_string()],
+    };
+
+    let fee_splitter_instance = app
+        .instantiate_contract(
+            code_id,
+            Addr::unchecked(sender),
+            &msg,
+            &[],
+            "fee-splitter",
+            None,
+        )
+        .unwrap();
+
+    let config_response: Config = app
+        .wrap()
+        .query_wasm_smart(fee_splitter_instance, &QueryMsg::Config {})
+        .unwrap();
+
+    assert_eq!(config_response.addresses, vec![first_tupple, second_tuple]);
+}
+
+#[test]
+#[should_panic(expected = "Provided weights exceed maximum allowed value")]
+fn fails_to_init_because_weights_not_correct() {
+    let mut app = CoreumApp::default();
+
+    let code_id = store_fee_splitter_code(&mut app);
+    let sender = "addr0000";
+
+    let first_tupple = ("tokenA".to_string(), Decimal::from_ratio(2u128, 1u128));
+    let second_tuple = ("tokenB".to_string(), Decimal::from_ratio(2u128, 1u128));
+    let msg = InstantiateMsg {
+        addresses: vec![first_tupple.clone(), second_tuple.clone()],
+        cw20_contracts: vec!["cw20_contract_one".to_string()],
+    };
+
+    app.instantiate_contract(
+        code_id,
+        Addr::unchecked(sender),
+        &msg,
+        &[],
+        "fee-splitter",
+        None,
+    )
+    .unwrap();
+}
+
+#[test]
+fn should_send_tokens_in_correct_amount() {
+    let mut deps = mock_coreum_deps(&[]);
+
+    deps.querier.with_balance(&[(
+        &String::from(MOCK_CONTRACT_ADDR),
+        &[
+            Coin {
+                denom: "ATOM".to_string(),
+                amount: Uint128::new(100_000000000000000000),
+            },
+            Coin {
+                denom: "TIA".to_string(),
+                amount: Uint128::new(100_000000000000000000),
```

Comment:
Format those numbers correctly. Also you can use smaller numbers to see the actual amount easier.